src/main/java/seedu/address/logic/commands/BatchDeleteCommand.java

```diff
@@ -52,19 +53,16 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
         model.updateFilteredPersonList(predicate);
-        List<Person> lastShownList = model.getFilteredPersonList();
+        List<Person> lastShownList = new ArrayList<>(model.getFilteredPersonList());
 
         StringBuilder feedbackToUser = new StringBuilder();
 
-        Person personToDelete;
-        int size = lastShownList.size();
-        for (int i = 0; i < size; i++) {
-            personToDelete = lastShownList.get(0);
+        for (Person person: lastShownList) {
             feedbackToUser.append(String
                     .format(MESSAGE_BATCH_DELETE_EACH_PERSON_SUCCESS,
-                            Messages.format(personToDelete))
+                            Messages.format(person))
             );
-            model.deletePerson(personToDelete);
+            model.deletePerson(person);
```

Comment:
I think case of the tag not being found is not handled/no error message is shown to the user?